Command/DumpCommand.php

```diff
@@ -0,0 +1,190 @@
+     */
+    protected function execute (InputInterface $input, OutputInterface $output)
+    {
+        $dumper       = $this->getContainer()->get('becklyn_mysql_dump.services.dump');
+        $dumperConfig = $this->getContainer()->get('becklyn_mysql_dump.services.configuration');
+
+        $connections = $dumperConfig->getDatabases($input->getOption('connections'));
+        $backupPath  = $dumperConfig->getBackupDirectory($input->getOption('path'));
+
+        /** @var FormatterHelper $formatter */
+        $formatter = $this->getHelper('formatter');
+
+        // Print headline
+        $headline = $formatter->formatBlock(array('', '  Becklyn MySQL Database Dumper', ''), 'comment');
+        $output->writeln($headline);
+
+        // If there are no database connection information available we can't dump anything
+        if (empty($connections))
+        {
+            $error = $formatter->formatBlock(array('', '  No connection data found.  ', ''), 'error');
+            $output->writeln("\n{$error}\n");
+
+            return 1;
+        }
+
+        $unresolvedConnections = false;
+        $dbBackupPaths         = [];
+        $tableRows             = [];
+
+        /** @var Connection $connectionData */
+        foreach ($connections as $connectionName => $connectionData)
+        {
+            // Check if the connection could not be resolved.
+            if ($connectionData === null)
+            {
+                $unresolvedConnections = true;
+
+                $tableRows[] = [
+                    '<error>- unresolved -</error>', $connectionName, '-'
+                ];
+
+                continue;
+            }
+
+            // Preserve the designated file names for the actual backup process as an actual dump
+            // may take very long the actual file names would differ from the one we printed to the user
+            $dbBackupPaths[$connectionName] = $this->getBackupFilePath($backupPath, $connectionName, $connectionData);
+
+            $tableRows[] = [
+                $connectionData->getDatabase(),
+                $connectionName,
+                $dbBackupPaths[$connectionName]
+            ];
+        }
+
+        // Print a nice table with the database name and the actual target file path
+        $this->getHelper('table')
+             ->setHeaders(array('Database', 'Connection', 'Backup file'))
+             ->setRows($tableRows)
+             ->render($output);
+
+        // Check if any connections could not be resolved and then print an error to abort.
+        if ($unresolvedConnections)
+        {
+            $error = $formatter->formatBlock(array('', 'Could not resolve one or more connections.', ''), 'error');
+            $output->writeln("\n{$error}\n");
+
+            return 1;
+        }
+
+
+        if (!$input->getOption('force'))
+        {
+            $output->writeln('');
+            $question = new ConfirmationQuestion('<question>Would you like to backup now? [Yn]</question>', true);
```

Comment:
Without the [Yn]?

I find them somewhat useful so you know what you need to enter. Also the capital letter shows which option it defaults to if you hit Enter. From what I see around in the Unix world it's kind of a standard (or maybe just a convention?).

In the end I wouldn't mind dropping the [Yn] part.